csv_export_invoice/models/csv_export_invoice.py

```diff
@@ -127,16 +127,18 @@ def export_order(invoice_line):
     def _get_line_amounts(self, line):
         """
 
-        :param line_amount: float, including taxes
-        :param tax: float, line tax amount
+        :param line: account.invoice.line record
         :return: a tuple of total amount, base_amount and tax_amount.
           base_amount and tax_amount are rounded to two decimals.
           total_amount is the sum of rounded base_amount and tax_amount
           for lines in the invoice.
         """
         taxes = line.invoice_line_tax_ids
+        # from src/odoo/addons/account/models/account_invoice.py
+        # is there a way not to copy paste this code ?
+        price = line.price_unit * (1 - (line.discount or 0.0) / 100.0)
```

Comment:
I had to do this in v9 to make sur the taxes were localized (ie. a 21% tax on a product converts to 0% intracomm in the case of a sale in EU).

Maybe the fields (`invoice_line_tax_ids` ?) in the account.invoice.line are better in v12. You'd need to look at function `_compute_price` which seems to take the discounts and taxes into account when setting

- price_subtotal
- price_total
- price_subtotal_signed

Reply:
Not sure we understood each other, the comment was about the line `price = line.price_unit * (1 - (line.discount or 0.0) / 100.0)` that I copied from the _compute_price method :

https://github.com/OCA/OCB/blob/daa274eb48324d563e1096e334a5a73034be71dc/addons/account/models/account_invoice.py#L1695